client/rpc-api/src/state/mod.rs

```diff
@@ -144,6 +145,10 @@ pub trait StateApi<Hash> {
 		at: Option<Hash>,
 	) -> FutureResult<Vec<StorageChangeSet<Hash>>>;
 
+	/// Returns proof of storage entries at a specific block's state.
+	#[rpc(name = "state_getReadProof")]
+	fn read_proof(&self, keys: Vec<StorageKey>, hash: Hash) -> FutureResult<StorageProof>;
```

Comment:
```suggestion
	fn read_proof(&self, keys: Vec<StorageKey>, hash: Option<Hash>) -> FutureResult<StorageProof>;
```

hash should be optional and default to the latest block as per all other methods.

Reply:
Proof must be verified using the state root of the block that generated the proof. The latest block keeps changing. One user uses this interface to get a proof in "the latest block", and then uses the state root in "the latest block" to verify it, which is error-prone. So I think it is necessary to specify a block.

Reply:
I'd still prefer to have that argument optional, the reason is consistency with other APIs and less queries if you are interested in the proof from latest state. The issue you mention is just verification and IMHO the right solution is to include the block hash (or other details) inside the response object of this query.

Reply:
Updated.